### PR TITLE
fix: #15931 incorrect recognition of dialing code

### LINKED ISSUE #15931: Incorrect recognition of phone number
### Issue Summary

This issue is related to #7370, however, that is a self-hosted cal, this bug is on cal.com directly.

When autofilling a 480 number without a country prefix, the default country shows US, but then it switches to a different country. This is not satisfactory for my needs. If there isn't a plus on the front, it should keep the +1 and append the autofilled phone number.

### Steps to Reproduce

1. Put 4805551234 into your autofill in chrome

2. Select that autofill

Screenshot 2024-07-26 at 11 41 50 AM

3. Behold, a different country shows up.

Screenshot 2024-07-26 at 11 41 56 AM

### Expected behavior

I should keep the +1 in front of the autofilled phone number if that phone number doesn't have a plus.

### Other information

_No response_

### Screenshots

_No response_

### Environment

- [X] app.cal.com
- [ ] Self-hosted Cal

### Desktop (please complete the following information)

```markdown
- OS: MacOS
- Browser: Chrome
- Version: Version 126.0.6478.183 (Official Build) (arm64)
```

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

- Fixes #15931 
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

**Root cause for bug:**
1. When any number is input , `<PhoneInput` component tries to autodetect first few numbers as dialing-code.
2. Usually browser autofill does not contain or does not insert phone number with dialing code, also browser autofill works when the input field is empty (so user removes prefilled dialing code if any and inserts phone number from autofill). Hence phone number without dialing code may get input.
3. The `<PhoneInput` component detects these first few numbers inserted (without dialing code ) as dialing code and selects matching country which may not be intended one.

**Solution Approach implemented**
1. The autoGuessing of Country is disabled in `<PhoneInput`
2. Thus Allowing user to select the Country instead of `<PhoneInput` autodetecting it.
3. Also by default , when a country is selected `<PhoneInput` will not prefix dialing code if some text already exists in input field. So this is customized to prefix dialing code if the input number is Not a valid phone number only. If user had input phone number with dialing-code , this is valid phone number and additional prefix does not happen.

Note: This will work regardless of default Country selection present or not.

**Before Changes :**

https://www.loom.com/share/2dea6738c2834e81936c5cd8baf107c1?sid=fb3a30f3-7aad-4591-bc61-c6876d42f9fd

**After Changes :**

https://www.loom.com/share/7b1c85b6e7ec4a5d99db4878dbbe87ac?sid=1fd90c22-b8e6-4146-97d8-57b4c779f86e

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Create a browser autofill address with phonenumber and refer the before changes video.

### CODE REVIEW

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Thanks for the PR @vijayraghav-io ! 

But this forces the country code selected via dropdown. 
If I paste a number without country code, I cannot simply add +1 or +91 in front of it. In fact it does not let me add '+' in the phone number. It just moves the cursor to next number.

https://github.com/user-attachments/assets/24ed366c-8265-428a-90b4-b23962bb41c3


This is what it does on main now. I can change the country code without having to select via drop down.


https://github.com/user-attachments/assets/2a70f4f9-0ec0-44b7-b135-b6e185de7312

**Comment by Amit91848** [comment]
I am not even sure if we need this anymore since we have removed default country code. If user pastes with country code then the country gets selected automatically, if not then the user can add their desired country code. Lmk what you think

File: packages/ui/form/PhoneInput.tsx [hunk 1]
```diff
@@ -18,6 +19,36 @@ export type PhoneInputProps = {
...
+    if (
+      !!value &&
+      !!selectedDialCode &&
+      value !== selectedDialCode &&
+      !!parsedDialCode &&
+      selectedDialCode !== parsedDialCode &&
+      !isValidPhoneNumber
+    ) {
+      updatedValue = `+${selectedDialCode}${value}`;
+    }
```
Comment (anikdhabal) [inline packages/ui/form/PhoneInput.tsx:47]: Currently, we don't need to prefix any country code in any case; we want the user to add the full phone number, including the country code.

File: packages/ui/form/PhoneInput.tsx [hunk 2]
```diff
@@ -52,6 +83,7 @@ function BasePhoneInput({ name, className = "", onChange, value, ...rest }: Phon
         marginLeft: "-4px",
       }}
       dropdownStyle={{ width: "max-content" }}
+      disableCountryGuess={true}
```
Comment (anikdhabal) [inline packages/ui/form/PhoneInput.tsx:86]: This appears to be correct and currently we need

**Comment by Amit91848** [comment]
> This resolves wrong detection of country (when user has not inserted dialing code) and inserts from browser autofill. Yes, user has to select the required country from dropdown, since autodetecting country is disabled.

We want the system to autodetect the country code based on the input.

The main issue was that when we had a default country code like +1, the autofill feature would replace the +1 with the autofilled input and use its starting digits to detect the country code, changing it altogether.

However, since we have removed the default country code in #15991, this issue is no longer relevant.

If we disable autodetection of the country code on input, users would have to manually select the country code from the dropdown, adding unnecessary extra steps.


Thanks for your efforts, @vijayraghav-io. I will be closing this PR.

**Comment by vijayraghav-io** [comment]
Still wrong detection happens, Yeah but if we are allowing for wrong detection and relying on user to manually reenter the dialing code, then this is not relevant.

**Comment by Amit91848** [comment]
Yes the whole point of removing default country code was for user to fill it. If user enters a number without country code either by using autofill or doing it manually, the issue is on user's end and not ours.

**Comment by vijayraghav-io** [comment]
Yeah i agree, My thought process or attempt was if we can fix this issue, may be we could bring back default country code feature. Also browser autofill is widely used feature instead of manual entry. Agreed disabling auto detecting has its downsides, also there is no other way currently to fix this currently.
Thankyou!  @Amit91848  for the review.

**Pull request closed without merging**